pages/specification/json-hyper-schema/_index.md

```diff
@@ -13,6 +13,531 @@ In essence:
+
+Keep following links to discover more content.
+
+### Actions
+
+#### Search
+
+Using a link to construct a URI from user input. (Analogous to `<form method="get">`)
+
+```json
+{
+  "links": [
+    {
+      "href": "/books"
+      "rel": "search",
+      "hrefSchema": {
+        "type": "object",
+        "properties": {
+          "title": { "type": "string" },
+          "author": { "type": "integer" },
+          ...
+        }
+      }
+    }
+  ]
+}
+```
+
+```http
+GET https://example.com/books?title=Moby-Dick
+```
+
+_This `GET` request searches for books with the title `Moby-Dick`._
+
+```http
+200 OK
+Content-Type: application/json
+Link: <https://example.com/schemas/books>; rel="describedby"
+
+[
+  {
+    "title": "Mobby-Dick",
+    "authorId": 100,
+    ...
+  }
+]
+```
+
+_The response returns a successful `200 OK` JSON-formatted data with a `Link` to the schema describing the book collection, and the response body contains an array of books, each with details such as the `title` and `authorId`._
+
+#### Submission
+
+Using a link to submit user input. (Analogous to `<form method="post">`)
+
+```json
+{
+  "links": [
+    {
+      "href": "/books"
+      "rel": "collection",
+      "submissionSchema": {
+        "type": "object",
+        "properties": {
+          "title": { "type": "string" },
+          "authorId": { "type": "integer" },
+          ...
+        },
+        "required": ["title", "authorId", ...]
+      }
+    }
+  ]
+}
+```
+
+```http
+POST https://example.com/books
+Content-Type: application/json
+
+{
+  "title": "Moby-Dick",
+  "authorId": 100,
+  ...
+}
+```
+
+_This `POST` request submits data to create a new book with the title "Moby-Dick" and `authorId` 100._
+
+```http
+204 No Content
+```
+
+_The request was successful, and the book was created, but the response contained no content._
+
+##### Submission Media Type
+
+You can also make submission will media types other than JSON.
+
+```json
+{
+  "links": [
+    {
+      "href": "/books"
+      "rel": "collection",
+      "submissionMediaType": "application/xml",
+      "submissionSchema": {
+        "type": "object",
+        "properties": {
+          "title": { "type": "string" },
+          "authorId": { "type": "integer" },
+          ...
+        },
+        "required": ["title", "authorId", ...]
+      }
+    }
+  ]
+}
+```
+
+```http
+POST https://example.com/books
+Content-Type: application/xml
+
+<Book>
+  <title>Moby-Dick<title>
+  <authorId>100</authorId>
+  ...
+</Book>
+```
+
+```http
+204 No Content
+```
+
+_This code shows using application/xml, to create a new book resource with a `POST` request._
+
+### Relations
+
+#### Registered relation
+
+A registered relation pointing to an author.
+
+```json
+{
+  "links": [
+    {
+      "href": "https://example.com/authors/100",
+      "rel": "author"
+    }
+  ]
+}
+```
+
+#### Custom relation
+
+A custom relation linking to a specific review.
```

Comment:
suggestion: "The following is an example of a custom relation linking to a specific review."